server/server.go

```diff
@@ -189,6 +190,28 @@ func (s Server) ListenAndServe() {
 			if authzhandler.Store == nil || err != nil {
 				klog.Fatalf("Error in initalizing cache. Error:%s", err.Error())
 			}
+
+			if s.AuthzRecommendedOptions.Azure.FetchListOfResources {
+				clusterType := ""
+
+				switch s.AuthzRecommendedOptions.Azure.AuthzMode {
+				case "arc":
+					clusterType = azureutils.ConnectedClusters
+				case "aks":
+					clusterType = azureutils.ManagedClusters
+				case "fleet":
+					clusterType = azureutils.Fleets
+				default:
+					klog.Fatalf("Authzmode %s is not supported for fetching list of resources", s.AuthzRecommendedOptions.Azure.AuthzMode)
+				}
+
+				operationsMap, err := azureutils.FetchListOfResources(clusterType, s.AuthRecommendedOptions.Azure.Environment, s.AuthzRecommendedOptions.Azure.AKSAuthzTokenURL, s.AuthzRecommendedOptions.Azure.KubeConfigFile, s.AuthRecommendedOptions.Azure.TenantID, s.AuthRecommendedOptions.Azure.ClientID, s.AuthRecommendedOptions.Azure.ClientSecret)
```

Comment:
should we make sure guard pod is not ready before this list is retrieved?

Reply:
added a readiness probe with /readyz